### ISSUE TITLE
Consider where to adopt records

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-boot#28155]
We could use records for many of our configuration property classes. A number of classes that are used as DTOs by the actuator endpoints could probably also becomes records. https://github.com/spring-projects/spring-boot/issues/28074 is a prerequisite for using records for configuration properties.

I've labelled this as a task for now. It might become an epic or an enhancement depending on what we decide.

**Comment by Mamun-Al-Babu-Shikder** [comment]
Yes, we can change massive amount of classes those are annotated with @ConfigurationProperties are converted to Java record. Sometime we need to declare static nested class inside configuration class, it is nice to fit record because record is static by default when we declare inside another class or record.
